### DISCUSSION TITLE
ParaView, VisIt export issues

### DISCUSSION THREAD

**Discussion by moroz3339** [mfem/mfem#3228] [Q&A]
Hello everyone!
I am trying to solve the linear elasticity problem by mfem. I have got displacement and further evaluate von mises stress, for that, I save displacement in an object of a class inherited from mfem::coefficient where I have redefined method mfem::coeffiicient::eval() to compute von Mises stress. And I call mfem::GridFunction::ProjectCoefficient() for to compute von Mises. I had exported it (von Mises) with mfem::VisItDataCollection and mfem::ParaViewDataCollection and I got several negative von Mises stress despite that in a vector gotten as result mfem::GridFunction::HostRead() doesn't exist any negative stresses. So, may you have suggestions what is wrong?

**Comment by vladotomov** [comment]
What is the order of the finite element function? Interpolation of high-order functions, with nonnegative dof values, can produce negative values.

**Reply by moroz3339** [reply]
I use a first-order function 
actually, I build finite element space like that:
mfem::L2_FECollection fec(order = 1, dim = 3)
mfem::FiniteElementSpace fespace(mesh, fec, dim=1)
further I describe mfem::GridFunction von_mises(fespace) and
I project a displacement contained in my inherited class Coefficient with overloaded Coefficient::eval() on von_mises. For it I use mfem::GridFunction::ProjectCoefficient.

**Reply by vladotomov** [reply]
I can't think of a reason to get negatives. If you send me a reproducer (code that projects the `Coefficient` to a `GridFunction` and exports to Visit, and a mesh), I can take a look.
